services/terrain/src/terrain/clients/data_info/raw.clj

```diff
@@ -97,6 +97,15 @@
 
 ;; CREATE
 
+(defn upload-file
+  [user dest-path filename content-type istream]
+  (http/post (str (url/url (cfg/data-info-base) "data"))
+             {:query-params {:user user
+                             :dest dest-path}
+              :multipart [{:part-name "file"
+                           :name filename
```

Comment:
`:part-name` defines what key it ends up under, but `:name` defines the filename that'll be used. This is not at all documented in clj-http. Multipart requests are annoying.